### ISSUE TITLE
about loss of training stage2 transformers

### ISSUE THREAD

**Issue by CrossLee1** [CompVis/taming-transformers#212]
Dear authors and everyone

I'm trying to reproduce the training of stage2 transformers, but got a very high loss, about 3~4 ..

could anyone provide your training curves on stage2 for reference ?

Thanks a lot ~

**Comment by nicolasfischoeder** [comment]
I am also interested in that, as I encounter the same phenomenon. Is that normal?

**Comment by nicolasfischoeder** [comment]
@CrossLee1  have you figured it out by now maybe?

@pesser @rromb Can you maybe comment on that?  :)

**Comment by order-a-lemonade** [comment]
my loss is even higher, about 6 (train on ffhq256). and the loss cant go down. Have you figured it out?
![image](https://github.com/CompVis/taming-transformers/assets/66902842/406528d2-50fd-4695-b225-225d302ce14a)

**Comment by robertchen245** [comment]
Loss lower than 5 is good enough to generate reasonable images. Mine is 4.6 on CelebAHQ.
Have you tried to sample some images?

**Comment by order-a-lemonade** [comment]
![image](https://github.com/CompVis/taming-transformers/assets/66902842/fe8d4378-3c00-44cc-9748-e31934aedac0)
i have forgotten what's the problem here(seems like that i edited the source code and get some bugs), but my training loss is normal now.

**Comment by robertchen245** [comment]
> ![image](https://private-user-images.githubusercontent.com/66902842/292826131-fe8d4378-3c00-44cc-9748-e31934aedac0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM1NzYzNDUsIm5iZiI6MTcwMzU3NjA0NSwicGF0aCI6Ii82NjkwMjg0Mi8yOTI4MjYxMzEtZmU4ZDQzNzgtM2MwMC00NGNjLTk3NDgtZTMxOTM0YWVkYWMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI2VDA3MzQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhN2JkYjg1MmIzZDlhZjVlNDMxZTJiMjJmNzUyZDA3MTNjNjgyNDY1YjBmOWQ1YTJjNzhiMzU1ZDU3ODZjODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IAGX0lY9fmlOVE3ZRB30aPZB2Cw_gxHO4NkxYEBLEGg) i have forgotten what's the problem here(seems like that i edited the source code and get some bugs), but my training loss is normal now.

You mentioned that you modify the lr to 4.5e-6. I wonder how many epoch and the batch_size. Though my sampled images looks okay, I want it much smaller 😂

**Comment by order-a-lemonade** [comment]
> > ![image](https://private-user-images.githubusercontent.com/66902842/292826131-fe8d4378-3c00-44cc-9748-e31934aedac0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM1NzYzNDUsIm5iZiI6MTcwMzU3NjA0NSwicGF0aCI6Ii82NjkwMjg0Mi8yOTI4MjYxMzEtZmU4ZDQzNzgtM2MwMC00NGNjLTk3NDgtZTMxOTM0YWVkYWMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI2VDA3MzQwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhN2JkYjg1MmIzZDlhZjVlNDMxZTJiMjJmNzUyZDA3MTNjNjgyNDY1YjBmOWQ1YTJjNzhiMzU1ZDU3ODZjODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IAGX0lY9fmlOVE3ZRB30aPZB2Cw_gxHO4NkxYEBLEGg) i have forgotten what's the problem here(seems like that i edited the source code and get some bugs), but my training loss is normal now.
> 
> You mentioned that you modify the lr to 4.5e-6. I wonder how many epoch and the batch_size. Though my sampled images looks okay, I want it much smaller 😂

I am not sure where did i mentioned about modifying the lr🤣. But my batch_size is 1. btw, i don't think longer training step is helpful for sample quality, because the generated images of my checkpoint in 400k step are not obviously better than the generated images of the checkpoint provided by authors which is in 13750 step.

**Comment by lxxie298** [comment]
> ![image](https://private-user-images.githubusercontent.com/66902842/292826131-fe8d4378-3c00-44cc-9748-e31934aedac0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0MzM1ODksIm5iZiI6MTcxNjQzMzI4OSwicGF0aCI6Ii82NjkwMjg0Mi8yOTI4MjYxMzEtZmU4ZDQzNzgtM2MwMC00NGNjLTk3NDgtZTMxOTM0YWVkYWMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDAzMDEyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwYTBhNDNjZThlNTFmM2FiMDAwMDBlYjQ0OTE1NGIzY2IzYjIwMWQ1NzUyYTA5ZGRmMWMwODM2YmMyZmNkNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5gV27zEPu_N7QJngqvBmeFWoePJUe3loZtKF32y7Gf8) i have forgotten what's the problem here(seems like that i edited the source code and get some bugs), but my training loss is normal now.

I came across your comment about resolving the issue with the loss not decreasing. It's impressive that you managed to figure it out! I'm currently facing a similar problem and would be incredibly grateful for any guidance you could provide. If it's not too much trouble, could you please share the changes you made or perhaps your training configuration file?😂 @order-a-lemonade
